### PR TITLE
Use cached underlyingMatchType when normalizing applied match aliases

### PR DESCRIPTION
**Pull request by EugeneFlesselle**
Based on #20168

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/Types.scala [hunk 1]
```diff
@@ -4647,35 +4645,22 @@ object Types extends TypeUtils {
...
      */
     override def underlyingMatchType(using Context): Type =
       if ctx.period != validUnderlyingMatch then
         validUnderlyingMatch = if tycon.isProvisional then Nowhere else ctx.period
         cachedUnderlyingMatch = superType.underlyingMatchType
       cachedUnderlyingMatch
 
-    override def tryNormalize(using Context): Type = tycon.stripTypeVar match {
-      case tycon: TypeRef =>
-        def tryMatchAlias = tycon.info match
```
Comment (dwijnand) [inline compiler/src/dotty/tools/dotc/core/Types.scala:4659]: The use of a local method is intentional, to keep methods small for JIT.  So you might want to keep that.

Comment (EugeneFlesselle) [reply compiler/src/dotty/tools/dotc/core/Types.scala:4659]: Does it need be within a `case TypeRef` to be optimal ?

Comment (dwijnand) [reply compiler/src/dotty/tools/dotc/core/Types.scala:4659]: No, it just needs to have a small body.  Having it super local just gives you handy references, so you don't have to lift them yourself into method parameters and arguments.

**Comment by EugeneFlesselle** [comment]
test performance please

**Pull request closed without merging**